pkg/blockatlas/tx.go

```diff
@@ -452,3 +456,97 @@ func GetEthereumTokenTypeByIndex(coinIndex uint) TokenType {
 	}
 	return tokenType
 }
+
+func (t Tx) AssetModel() (models.Asset, bool) {
+	var asset models.Asset
+	switch t.Meta.(type) {
```

Comment:
I have seen very similar code with memo implementation for spam, can it be reused?

Reply:
Good idea, it is possible to make a common function to parse data like from, to,symbol and etc and re-use in 1-3 places